### DISCUSSION TITLE
MET issue installing on new Intel Compilers

### DISCUSSION THREAD

**Discussion by HathewayWill** [dtcenter/METplus#2165] [Installation] [answered]
Good afternoon,

I am having issues compiling and installing the MET and METPlus tools using the new intel compilers.
I have been following these guides for porting the old compilers (ifort, icc, etc) to the new compilers (ifx, icx, etc)
https://www.intel.com/content/www/us/en/developer/articles/guide/porting-guide-for-ifort-to-ifx.html
https://www.intel.com/content/www/us/en/developer/articles/guide/porting-guide-for-icc-users-to-dpcpp-or-icx.html

For reference the new intel compilers uses these options:

```
export CC=icx
export CXX=icpx
export FC=ifx
export F77=ifx
export F90=ifx
export MPIFC="mpiifort -fc=ifx"
export MPIF77="mpiifort -fc=ifx"
export MPIF90="mpiifort -fc=ifx"
export MPICC="mpiicc -cc=icx"
export MPICXX="mpiicpc -cxx=icpx"
```

I have attached all the log files and output that I have grabbed from the compilation.

[compile_MET_all.sh.log.txt](https://github.com/dtcenter/METplus/files/11392097/compile_MET_all.sh.log.txt)

[compile_MET_all.sh.txt](https://github.com/dtcenter/METplus/files/11392098/compile_MET_all.sh.txt)

[DTC_NEW_INTEL.sh.txt](https://github.com/dtcenter/METplus/files/11392099/DTC_NEW_INTEL.sh.txt)

[compile_MET_all.sh.txt](https://github.com/dtcenter/METplus/files/11392266/compile_MET_all.sh.txt)

[buf.make.log.txt](https://github.com/dtcenter/METplus/files/11392101/buf.make.log.txt)

[gsl.make.install.log.txt](https://github.com/dtcenter/METplus/files/11392269/gsl.make.install.log.txt)

[gsl.make.log.txt](https://github.com/dtcenter/METplus/files/11392270/gsl.make.log.txt)

[jasper.config.log.txt](https://github.com/dtcenter/METplus/files/11392111/jasper.config.log.txt)

[jasper.configure.log.txt](https://github.com/dtcenter/METplus/files/11392112/jasper.configure.log.txt)

[libpng.make.log.txt](https://github.com/dtcenter/METplus/files/11392113/libpng.make.log.txt)

[libpng.make_install.log.txt](https://github.com/dtcenter/METplus/files/11392114/libpng.make_install.log.txt)

[zlib.make.log.txt](https://github.com/dtcenter/METplus/files/11392121/zlib.make.log.txt)

[zlib.make_install.log.txt](https://github.com/dtcenter/METplus/files/11392122/zlib.make_install.log.txt)

**Comment by jprestop** [comment]
Hi @HathewayWill.  Looking at the screen output from the attached compile_MET_all.sh.log.txt file, it looks like the compilation stopped with jasper:
```
Compiling JASPER at Tue May  2 10:39:33 AM CDT 2023
cd /home/workhorse/DTC_Test/MET-11.0.2/external_libs/jasper/jasper-1.900.1
./configure --prefix=/home/workhorse/DTC_Test/MET-11.0.2/external_libs > configure.log 2>&1
make > make.log 2>&1
make returned with non-zero (2) status
```
The jasper config.log file seems to have successfully run.  Can you please pass along the make.log file from jasper?

Also, can you please tell me if you have a bufrlib library file in /home/workhorse/DTC_Test/MET-11.0.2/external_libs/lib?

**Comment by HathewayWill** [comment]
![image](https://user-images.githubusercontent.com/118180342/236352719-df76650e-35df-4abd-82db-bab85b952f4d.png)

[jasper.make.log.txt](https://github.com/dtcenter/METplus/files/11402379/jasper.make.log.txt)

Suggestion @jprestop Since there are so many libraries with the same name file like "make.log" "config.log" "configure.log" "make.install.log" would it make sense to rename each libraries log files like I have named them in my uploads?  "<name_of_library>.make.log" "<name_of_library>.make.install.log" "<name_of_library>.config.log" "<name_of_library>.configure.log" ?

That way if a user has to upload the files to a help ticket it is automatically known which library it is being downloaded from.

**Reply by jprestop** [reply]
Hi @HathewayWill.  I just wanted let you know that I am working on a solution for you to test.  I will follow up once it's ready.

**Comment by jprestop** [comment]
HI @HathewayWill.  Thanks for the suggestion to rename the log files in the script.  I have created an issue to enhance our compilation script: [Enhance compile_MET_all.sh to include the library/app name in the log filename #2540](https://github.com/dtcenter/MET/issues/2540)

I see the problem in the jasper.make.log file:

> jas_getopt.c:129:7: error: call to undeclared function 'jas_eprintf'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]

I thought that trying a newer version of jasper might work better since implicit function declarations are not supported by ISO C99 and later, however, newer versions of jasper require cmake.  So, if you don't have cmake on your computer you would need to download and install that.  I tried grabbing both jasper-4.0.0 and jasper-3.0.5, using a machine where I have access to Intel compilers (but not these latest ones that you are using), but the version of cmake on the computer was too old, because I received the error "CMake 3.12 or higher is required.  You are running version 2.8.12.2".  I eventually landed on jasper-2.0.33.  

Please grab jasper-2.0.33.tar.gz:

> wget https://github.com/jasper-software/jasper/releases/download/version-2.0.33/jasper-2.0.33.tar.gz

to replace the current version of jasper in your tar_files directory.  And use the attached compile_MET_all.sh script (renamed compile_MET_all.sh.txt only so that I could upload it here - once you rename it to compile_MET_all.sh you may need to change the permissions - _chmod 775 compile_MET_all.sh_ to make it executable), which has updated commands to compile this new version of jasper using cmake, with your installation configuration file and give that a try.  

Please let us know how it goes.

[compile_MET_all.sh.txt](https://github.com/dtcenter/METplus/files/11454663/compile_MET_all.sh.txt)

**Reply by jprestop** [reply]
Hi @HathewayWill.  I just wanted to check in and see if you've had the opportunity to try this out yet.  If not, that's ok, there is no rush.

I also wanted to let you know that I've done the work for issue [Enhance compile_MET_all.sh to include the library/app name in the log filename #2540](https://github.com/dtcenter/MET/issues/2540) and included work for [Modify compile_MET_all.sh to allow user to specify Compiler and Python version to load #2485](https://github.com/dtcenter/MET/issues/2485) in the same branch.  It is in the [Pull Request review](https://github.com/dtcenter/MET/pull/2548) stage right now and should be included in the MET-5.1.0 release.  Thanks for the suggestion for improvement!

**Comment by HathewayWill** [comment]
Hello Julie

I will be out of touch for two months due to recovery from major spine
surgery.

**Reply by jprestop** [reply]
Hi @HathewayWill.  Thank you for letting us know.  Please take care of yourself.  We wish you a restful and quick recovery.

**Comment by georgemccabe** [comment]
Hi @HathewayWill , I hope everything went well with your recovery. Is there anything else needed to be done for this discussion or have you and @jprestop resolved this already?

**Reply by HathewayWill** [reply]
Good afternoon @georgemccabe .  @jprestop and I haven't had a chance to work on this one yet.  

The new Intel LLVM compilers are still actively being developed by intel and I am in talks with them with the bugs I have found when trying to use them for MET.  For now can we leave this thread open?

**Reply by georgemccabe** [reply]
Yes, we can leave this thread open. I just wanted to follow up to make sure you didn't need anything else from us. Let us know here when you have had a chance to work on it.

**Reply by jprestop** [reply, marked as answer]
@HathewayWill @georgemccabe Since @HathewayWill created a MET issue ([Bugfix: new Intel LLVM compilers not compatible with MET #2611](https://github.com/dtcenter/MET/issues/2611)) for this topic I think it would be best to keep any dialogue in the issue instead of here in the Discussion and close this discussion (since it is unresolved).  Any objections?

**Reply by HathewayWill** [reply]
None by me
